tests/platform_tests/sfp/util.py

```diff
@@ -32,6 +32,20 @@ def parse_eeprom(output_lines):
     return res
 
 
+def parse_eeprom_hexdump(output_lines):
```

Comment:
This should be more readable: 

def parse_eeprom_hexdump(data):
    # Define a regular expression to capture all required data
    regex = re.compile(
        r"EEPROM hexdump for port (\S+)\n"                     # Capture port name
        r"(?:.*?\n)*?"                                        # Match and skip intermediate lines
        r"((Lower|Upper) page \S+)\n"                         # Capture page type and identifier
        r"((?:[0-9a-fA-F]{8}(?: [0-9a-fA-F]{2}){16} .*\n)+)"  # Capture hex data block
    )
    
    # Dictionary to store parsed results
    parsed_data = {}
    
    # Find all matches in the data
    matches = regex.findall(data)
    for port, page_full, page_type, hex_data in matches:
        if port not in parsed_data:
            parsed_data[port] = {}
        
        # Parse hex data block into individual hex values
        hex_lines = hex_data.splitlines()
        hex_values = [
            value
            for line in hex_lines
            for value in line[9:56].split()  # Extract hex bytes from columns 9-56
        ]
        
        parsed_data[port][page_full] = hex_values
    
    return parsed_data